src/models/event-timeline-set.ts

```diff
@@ -522,16 +570,40 @@ export class EventTimelineSet extends TypedEventEmitter<EmittedEvents, EventTime
      * Add an event to the end of this live timeline.
      *
      * @param {MatrixEvent} event Event to be added
-     * @param {string?} duplicateStrategy 'ignore' or 'replace'
-     * @param {boolean} fromCache whether the sync response came from cache
+     * @param {IAddLiveEventOptions} options addLiveEvent options
      * @param roomState the state events to reconcile metadata from
      */
     public addLiveEvent(
         event: MatrixEvent,
-        duplicateStrategy: DuplicateStrategy = DuplicateStrategy.Ignore,
+        duplicateStrategy?: DuplicateStrategy,
+        fromCache?: boolean,
+        roomState?: RoomState,
+    ): void;
```

Comment:
Should we add deprecation warnings for these old overloads now? This way we have some pathway to cleaning up this technical debt in a year or two.

Reply:
Good idea

Reply:
Done ✅ https://github.com/matrix-org/matrix-js-sdk/pull/2299/commits/e4bd5188735e614a814f19ce4fa616243b052481

 1. Moved our preferred overload to the top so it's suggested first
 1. Added deprecation warnings in usage via `logger.warning`
 1. I've also added [`@deprecated`](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#deprecated) TypeScript decorations